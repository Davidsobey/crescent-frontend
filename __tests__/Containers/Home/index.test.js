import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store'; // Smart components

// Component to be tested
import HomeComponent from '../../../src/Containers/Home/index.js';


const mockStore = configureStore();
const initialState = {
  LoginReducer:{
    user: false,
  },
  UserReducer:{
    users: [],
    loading: false,
  },
  CourseReducer:{
    courses: [],
    loading: false,
  },
  ClientReducer:{
    userEnrolments: [],
    loading: false,
    subscriptions: [],
    subscriptions_loading: false,
  },
  PolicyReducer:{
    policyAcknowledgements: [],
    loading: false,
  },
  PaymentReducer:{
    paymentStatuses: [],
    loading: false,
  },
};
const store = mockStore(initialState);

describe('<HomeComponent />', () => {
  describe('render()', () => {
    test('renders the component', () => {
      const wrapper = shallow(<HomeComponent store={store} />);
      const component = wrapper.dive();

      expect(toJson(component)).toMatchSnapshot();
    });
  });
});