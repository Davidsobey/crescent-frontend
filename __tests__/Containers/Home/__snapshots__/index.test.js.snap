// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeComponent /> render() renders the component 1`] = `
<Connect(Form(HomeComponent))
  courses={Array []}
  courses_loading={false}
  destroyOnUnmount={true}
  dispatch={[Function]}
  enableReinitialize={false}
  forceUnregisterOnUnmount={false}
  form="homeComponent"
  getFormState={[Function]}
  keepDirtyOnReinitialize={false}
  paymentStatuses={Array []}
  paymentStatuses_loading={false}
  persistentSubmitErrors={false}
  policyAcknowledgements={Array []}
  policyAcknowledgements_loading={false}
  pure={true}
  shouldAsyncValidate={[Function]}
  shouldError={[Function]}
  shouldValidate={[Function]}
  shouldWarn={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  subscriptions={Array []}
  subscriptions_loading={false}
  touchOnBlur={true}
  touchOnChange={false}
  updateUnregisteredFields={false}
  user={false}
  userEnrolments={Array []}
  userEnrolments_loading={false}
  users={Array []}
  users_loading={false}
/>
`;
